### PR TITLE
[CX-2319]: Analyze impact of a given table

### PR DESCRIPTION
**Pull request by anatoly-scherbakov**
# Why

We'd like to illustrate how we can use Datafold GraphQL API to analyze downstream impact of a whole table or its individual column(s).

# What

```shell
j analyze-impact 1 LINEAGE_QA.CI.DB_COLUMNS --columns ID,NUMBER
```
⇒
image

### CODE REVIEW

File: analyze_impact.py [hunk 1]
```diff
@@ -36,72 +45,105 @@ def get_linked_columns(
...
-    parser = argparse.ArgumentParser(description='Analyze downstream dependencies')
-    parser.add_argument('data_source_id', type=int)
-    parser.add_argument('full_table_name', type=str,
-                        help='db.schema.name format, case-sensitive')
-    parser.add_argument('columns', metavar='column', type=str, nargs='*',
-                        help='column name to trace; ommit to trace all columns')
+    lineage_data = execute_stored_query(
+        QUERIES / 'lineage_analyze_impact.gql',
+        primaryUid=table_uid,
+        depthDownstream=1000,
```
Comment (fpopa) [inline analyze_impact.py:52]: Should we use a small value here for the default value?

Not sure who and why would have that level of depth, but even if we encounter a bug on our end, it's safer to have it lower.

default to 25 but allow up to 100?

File: analyze_impact.py [hunk 2]
```diff
@@ -36,72 +45,105 @@ def get_linked_columns(
...
+    Print downstream dependencies of a given table.
+
+    How to obtain Data Source ID:
+
+    * Open https://app.datafold.com,
+
+    * Go to **Settings** → Data Sources,
+
+    * Find your Data Source; the leftmost column in the Data Sources list is ID.
+    """
```
Comment (fpopa) [inline analyze_impact.py:138]: If I'd read this, the first thing I'd do is check that the data_sources are secured by trying a random data_source ID that's not part of the same org 😅 

I guess we do some cross-org checks for fetching this data, right?

File: analyze_impact.py (hunk 1 above)
Comment (anatoly-scherbakov) [reply analyze_impact.py:52]: This is for impact analysis so we have to traverse the whole depth of the graph. The value of 1000 was used here before me; in data diffs UI the front end uses the value of 99.

We might want to create a separate endpoint which would ensure to travel the whole depth — but we haven't done that yet.

Comment (fpopa) [reply analyze_impact.py:52]: I thought we'd help customers run this type of queries on their own? 🤔

File: analyze_impact.py (hunk 2 above)
Comment (anatoly-scherbakov) [reply analyze_impact.py:138]: Yes, that is a part of the backend.

**Review by fpopa — APPROVED** [review]

File: analyze_impact.py (hunk 1 above)
Comment (anatoly-scherbakov) [reply analyze_impact.py:52]: Yes, and we're considering building an API which will basically do what the post-processing in this repo does.

**Pull request merged**